src/btree/bt_read.c

```diff
@@ -8,6 +8,333 @@
 
 #include "wt_internal.h"
 
+#define	WT_MAX_PREFIX_SIZE						\
+    (sizeof(char) +							\
+    sizeof(uint32_t) + sizeof(uint8_t) + WT_BTREE_MAX_ADDR_COOKIE)
+
+/*
+ * __las_build_prefix --
+ *	Build the unique file/address prefix.
+ */
+static void
+__las_build_prefix(WT_SESSION_IMPL *session, uint32_t btree_id,
+    const uint8_t *addr, size_t addr_size, uint8_t *prefix, size_t *prefix_lenp)
+{
+	size_t len;
+	void *p;
+
+	/*
+	 * Build the page's unique key prefix we'll search for in the lookaside
+	 * table, based on the file's ID and the page's block address.
+	 */
+	p = prefix;
+	*(char *)p = WT_LAS_RECONCILE_UPDATE;
+	p = (uint8_t *)p + sizeof(char);
+	memcpy(p, &btree_id, sizeof(uint32_t));
+	p = (uint8_t *)p + sizeof(uint32_t);
+	*(uint8_t *)p = (uint8_t)addr_size;
+	p = (uint8_t *)p + sizeof(uint8_t);
+	memcpy(p, addr, addr_size);
+	p = (uint8_t *)p + addr_size;
+	len = sizeof(char) + sizeof(uint32_t) + sizeof(uint8_t) + addr_size;
+	WT_ASSERT(session, WT_PTRDIFF(p, prefix) == len);
```

Comment:
> Setting `key_format=bIuIQ` (I think?) would replace all of this code with a single `set_key` call.  By setting `counter=0` and `txnid=0`, you could get the same effect as calculating the key prefix.

It's more complicated than that because the row's key is included in the LAS key. I'll take a run at it, and see where I end up.